src/OAuth/OAuth2/Service/BattleNetCN.php

```diff
@@ -0,0 +1,18 @@
+<?php
+namespace OAuth\OAuth2\Service;
+
+use OAuth\Common\Http\Uri\Uri;
+use OAuth\Common\Consumer\Credentials;
+use OAuth\Common\Http\Client\ClientInterface;
+use OAuth\Common\Storage\TokenStorageInterface;
+use OAuth\Common\Http\Uri\UriInterface;
+
+class BattleNetCN extends BattleNetBase
+{
+    public function __construct(Credentials $credentials, ClientInterface $httpClient, TokenStorageInterface $storage, $scopes = array(), UriInterface $baseApiUri = null)
```

Comment:
This code is not repeating itself and it is not logically the same.  It is creating instantiable callers that can be used in a system where the library is called without the ability to pass further parameters (such as, PHPBB for which this modification was made).  

The instantiable classes are unnecessary if someone is willing and able to specify the region on the base class, however these classes allow a clean initialization that matches the behavior of the rest of the library if they are unable to do so.  The base class contains all common code to avoid duplication.

The one change that would probably be worth making is adding a constructor on the base class that takes region and sets the base url accordingly as the base url assembly is more arguably duplicate code.